### ISSUE TITLE
GenerationType Strategies not supported by ESClient

### ISSUE THREAD

**Issue by dreamingo88** [Impetus/kundera#845]
Entity:
`
public class Category {

```
@Id
@GeneratedValue(strategy = GenerationType.AUTO)
public String id;

@Required
@MinSize(3)
public String name;

@Required
@MinSize(3)
public String code;
```

}
`

Create data:
`for(int i = 1; i <= 50; i++) { 
Category ctg = new Category(); 
ctg.name = UUID.randomUUID().toString; 
ctg.code = "abc"; 
em.persist(ctg); 
}`

it will throw an exception:
`java.lang.UnsupportedOperationException: GenerationType Strategies not supported by this client : ESClient`

Which GenerationType can i use in ES? 
if none, is there any plan to implement it? 
thanks!

**Comment by karthikprasad13** [comment]
@dreamingo88 

We can take this up in future release if it's a critical requirement for you. Care to share your use case ?

-Karthik

**Comment by dreamingo88** [comment]
@karthikprasad13 
Very glad to share my use case, how to send it to you?

**Comment by chhavigangwal** [comment]
Please mail it to kundera@impetus.co.in

Chhavi

**Comment by dreamingo88** [comment]
@chhavigangwal @karthikprasad13 
I have mailed my use case to kundera@impetus.co.in, please check.

**Comment by chhavigangwal** [comment]
@dreamingo88 

I can't find the attachement

Chhavi

**Comment by karthikprasad13** [comment]
@dreamingo88 

Any updates?

-Karthik

**Comment by devender-yadav** [comment]
@dreamingo88 

Added support for `GenerationType.AUTO` strategy. It will be available from Kundera 3.6 onwards. 

-Dev
